File: JSPM.Web/Modules/ServerTypes/Organisation/BatchesForm.ts
import { IntegerEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface BatchesForm {
    DivisionId: IntegerEditor;
    BatchName: StringEditor;
}

export class BatchesForm extends PrefixedContext {
    static formKey = 'Organisation.Batches';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BatchesForm.init)  {
            BatchesForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(BatchesForm, [
                'DivisionId', w0,
                'BatchName', w1
            ]);
        }
    }
}

File: JSPM.Web/Modules/Organisation/Batches/BatchesForm.cs
using Serenity.ComponentModel;

namespace JSPM.Organisation.Forms;

[FormScript("Organisation.Batches")]
[BasedOnRow(typeof(BatchesRow), CheckNames = true)]
public class BatchesForm
{
    public int DivisionId { get; set; }
    public string BatchName { get; set; }
}